src/gui/torrentoptionsdialog.h

```diff
@@ -26,40 +26,64 @@
  * exception statement from your version.
  */
 
-#ifndef UPDOWNRATIODIALOG_H
-#define UPDOWNRATIODIALOG_H
+#pragma once
 
 #include <QDialog>
 
+namespace BitTorrent
+{
+    class InfoHash;
+}
+
 namespace Ui
 {
-    class UpDownRatioDialog;
+    class TorrentOptionsDialog;
 }
 
-class UpDownRatioDialog final : public QDialog
+class TorrentOptionsDialog final : public QDialog
 {
     Q_OBJECT
 
 public:
-    UpDownRatioDialog(bool useDefault, qreal initialValue, qreal maxValue,
-            int initialTimeValue, int maxTimeValue,
-            QWidget *parent = nullptr);
-    ~UpDownRatioDialog();
-
-    bool useDefault() const;
-    qreal ratio() const;
-    int seedingTime() const;
+    explicit TorrentOptionsDialog(QWidget *parent, const QVector<BitTorrent::InfoHash> selectedTorrents);
+    ~TorrentOptionsDialog();
 
 public slots:
     void accept() override;
 
 private slots:
+    void updateUploadSpinValue(int val);
+    void updateDownloadSpinValue(int val);
+    void updateUploadSliderValue(int val);
+    void updateDownloadSliderValue(int val);
+
     void handleRatioTypeChanged();
     void enableRatioSpin();
     void enableTimeSpin();
 
+    void disableTristateCheckboxDHT();
+    void disableTristateCheckboxPEX();
+    void disableTristateCheckboxLSD();
+
 private:
-    Ui::UpDownRatioDialog *m_ui;
+    void initDialog();
+
+    int getUploadSpeedLimit() const;
+    int getDownloadSpeedLimit() const;
+
+    bool useGlobalShareLimits() const;
+    qreal getRatio() const;
+    int getSeedingTime() const;
+
+    bool isDisableDHTChecked() const;
+    bool isDisablePEXChecked() const;
+    bool isDisableLSDChecked() const;
+
+    int m_globalUploadLimit;
+    int m_globalDownloadLimit;
+
+    QVector<BitTorrent::InfoHash> m_selectedTorrentsHashes;
```

Comment:
m_torrentHashes or m_infoHashes
"selected" is irrelevant in this context (because there are no other torrents for this class).